packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -203,8 +198,8 @@ export function initInput(
           // prematurely marking required inputs as invalid. Equality is compared
           // to the current value in case the browser provided value is not an
           // empty string.
-          if (isButton || value !== node.value) {
-            node.value = toString(value);
+          if (isButton || toString(getToStringValue(value)) !== node.value) {
```

Comment:
not sure if intentional but you added a toString() here

Reply:
Yea I think that's a bug fix.